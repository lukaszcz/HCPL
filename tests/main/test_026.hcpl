
// dyndef

dyndef y = 0;

let f = \x {
   dyndef y = y + x;
   if x == 0 then
      y
   else
      f (x - 1)
};

print (f 2);
print (f 10);
print (f 5);

{
   dyndef y = 16;
   print (f 2);
   print (f 3);
};

print (f 2);
print (f 3);
