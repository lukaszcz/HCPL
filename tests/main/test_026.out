3
55
15
19
22
3
6
